src/lib/bootloader/stage1.rb

```diff
@@ -60,27 +60,32 @@ def propose
     def propose_boot_location
       raise "Boot partition disk not found" if boot_partition_disk.empty?
       selected_location = :mbr
+      separate_boot = Yast::BootStorage.BootPartitionDevice != Yast::BootStorage.RootPartitionDevice
 
       # there was check if boot device is on logical partition
       # IMO it is good idea check MBR also in this case
       # see bug #279837 comment #53
       if boot_partition_on_mbr_disk?
-        separate_boot = Yast::BootStorage.BootPartitionDevice != Yast::BootStorage.RootPartitionDevice
         selected_location =  separate_boot ? :boot : :root
       elsif underlying_boot_partition_devices.size > 1
         selected_location = :mbr
       end
 
-      if logical_boot? && extended_partition &&
-          underlying_boot_partition_devices.size > 1
+      if logical_boot? && extended_partition
         selected_location = :extended
       end
 
-      if logical_boot_with_btrfs?
+      if boot_with_btrfs? && logical_boot?
         log.info "/boot is on logical parititon and uses btrfs, mbr is favored in this situration"
```

Comment:
I know this typo is not from your commit, but maybe you could fix it: `parititon` -> `partition` :)